File: dwh/java/Program7.java
package com.java.test_20_12_7;

public class Program7 {
    public static void main(String[] args) {
        int[] nums = {1,3,4,2,5,3};
        System.out.println(findNum(nums));
    }

    /**
     * 思路： 将数组中的所有元素累加，然后减去 正常从1-n的和
     *         得到的结果就是数组中多出来的那个元素。
     *         注意:nums.length 的大小为 n+1
     */
    private static int findNum(int[] nums){
        int num = 0;
        for (int i = 0; i < nums.length; i++){
            num += nums[i];
        }
        num = num - (nums.length - 1) * nums.length / 2;
        return num;
    }
}

File: carl/python/program_7.py
# for program_7.md
def find_repeat_num(nums:list)->int:
  if not nums or len(nums)<1:
    raise("nums Illegal！")
  res=0
  for num in range(len(nums)):
    if num == 0:
      res^=nums[num]
      continue
    res^=nums[num]
    res^=num
  return res
    
if __name__=='__main__':
  arr1=[1,3,4,2,5,3]
  arr2=[ x for x in range(1,1001)]
  arr2.append(5)
  print(find_repeat_num(arr1))
  print("-"*6)
  print(find_repeat_num(arr2))